src/worker/mod.rs

```diff
@@ -286,6 +304,23 @@ impl<S: AsyncBufRead + AsyncWrite + Send + Unpin, E: StdError + 'static + Send>
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+/// A reason why [`Worker::run`] has discontinued.
+pub enum RunCeaseReason {
```

Comment:
```suggestion
#[non_exhaustive]
pub enum StopReason {
```
As it feels likely that there may be more stop reasons in the future. Also, I think it's unlikely that callers will benefit from exhaustive matching on this `enum`.

Reply:
It is sad, though I also don't have a great alternative :sweat_smile: After all, if we instead have an `Other` variant, that'll more or less be equivalent to matching on `_` except that you're given the string name of the other variant (which you could otherwise get by invoking `Debug` or we could add a method for it). I suppose having an `Other` variant would also mean that if a new "officially supported" variant comes along, you'd be told your `match` is now missing an arm. However, that's also the source of the problem, namely that it'd require a breaking change and thus a new major version (which users then need to remember to adopt).